festim/hydrogen_transport_problem.py

```diff
@@ -123,11 +129,50 @@ def initialise(self):
         self.t = fem.Constant(self.mesh.mesh, 0.0)
         self.dt = self.settings.stepsize.get_dt(self.mesh.mesh)
 
+        self.define_temperature()
         self.define_boundary_conditions()
         self.create_formulation()
         self.create_solver()
         self.defing_export_writers()
 
+    def define_temperature(self):
+        """If temperature value is given only dependent on t, create a
+        fem.Constant else create an expression to be updated and a function"""
+        # if temperature is a float or int, create a fem.Constant
+        if isinstance(self.temperature, (float, int)):
+            self._temperature_fenics_value = F.as_fenics_constant(
+                self.temperature, self.mesh.mesh
+            )
+        elif isinstance(self.temperature, fem.Constant):
+            self._temperature_fenics_value = self.temperature
+
+        # if temperature is callable, process accordingly
+        elif callable(self.temperature):
+            arguments = self.temperature.__code__.co_varnames
+            if "t" in arguments and "x" not in arguments and "T" not in arguments:
+                # only t is an argument
+                self.temperature_fenics_value = F.as_fenics_constant(
+                    mesh=self.mesh.mesh, value=self.temperature(t=float(self.t))
+                )
+                self.temperature_time_dependent = True
```

Comment:
I feel like `temperature_time_dependent` should be a property of the class.

Something like:

```python
@property
def temperature_time_dependent(self):
    if self.temperature is None:
        return False
    if callable(self.temperature):
        arguments = self.temperature.__code__.co_varnames
        return "t" in arguments:
    else:
        return False